beacon_node/network/src/sync/network_context.rs

```diff
@@ -296,12 +296,25 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
         }
     }
 
+    /// Request block of `block_root` if necessary by checking:
+    /// - If the da_checker has a pending block from gossip or a previous request
+    ///
+    /// Returns false if no request was made, because the block is already imported
     pub fn block_lookup_request(
         &mut self,
         lookup_id: SingleLookupId,
         peer_id: PeerId,
-        request: BlocksByRootSingleRequest,
+        block_root: Hash256,
     ) -> Result<bool, &'static str> {
+        if self
+            .chain
+            .reqresp_pre_import_cache
+            .read()
+            .contains_key(&block_root)
+        {
+            return Ok(false);
+        }
```

Comment:
I think this is the cause of the recent de-sync issues.

Consider the following sequence of events:

1. Block arrives on gossip, we add it to the `reqresp_pre_import_cache`.
2. We send the block's payload to the EL for verification.
3. We receive blobs on gossip and add them to the DA cache.
4. The EL _times out_. Block processing is aborted as a result.
5. Subsequent attempts to look up the block and re-process it are blocked by the entry in the `reqresp_pre_import_cache`.
6. We stall and are unable to reprocess the block until range sync and/or a blocks by range request is made in spite of the check above.

I've confirmed that this happened for block 0xfc4a1afe39d84a410c61f213cf15cb405fdd15b96ca54f0c826d4054671b1275 at slot 1574856 on Holesky. We get the block and all the blobs, then the EL times out and everything gets stuck:

> May 04 05:31:20.803 DEBG Failed to verify execution payload      error: ExecutionPayloadError(RequestFailed(EngineError(Api { error: HttpClient(url: http://127.0.0.1:8551/, kind: timeout, detail: operation timed out) })))

Reply:
Actually scratch that, @jimmygchen pointed out that we remove from the `reqresp_pre_import_cache` on failure here:

https://github.com/sigp/lighthouse/blob/63fad7e330c4559474243346de7bc7389f6541d0/beacon_node/beacon_chain/src/beacon_chain.rs#L2945-L2950